src/kirby/components/button/button.component.scss

```diff
@@ -1,11 +1,65 @@
 @import '../../scss/web-imports';
 
 :host {
-  --kirby-button-border-radius: 2rem;
-  --kirby-button-background-color: #{get-color('primary')};
-  --kirby-button-border-color: #{get-color('primary')};
-  --kirby-button-color: #{get-color('primary-contrast')};
+  @mixin background-color($color) {
+    --kirby-button-background-color:#{get-color($color)};
+  }
+
+  @mixin border-color($color) {
+      --kirby-button-border-color: #{get-color($color)};
+  }
+
+  @mixin text-color($color) {
+      --kirby-button-color: #{get-color($color)};
+  }
+
+  :host-context(.primary).attention-level1,
```

Comment:
Lot of repetition - use sass loop over colors instead - see card.component.scss for example